src/Plugins/RestServer/Controllers/v1/LedgerController.cs

```diff
@@ -0,0 +1,390 @@
+        /// <response code="200">Successful</response>
+        /// <response code="400">An error occurred. See Response for details.</response>
+        [HttpGet("blocks", Name = "GetBlocks")]
+        [ProducesResponseType(StatusCodes.Status204NoContent)]
+        [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(Header[]))]
+        public IActionResult GetBlocks(
+            [FromQuery(Name = "page")]
+            uint skip = 1,
+            [FromQuery(Name = "size")]
+            uint take = 50)
+        {
+            if (skip < 1 || take < 1 || take > RestServerSettings.Current.MaxPageSize)
+                throw new InvalidParameterRangeException();
+            //var start = (skip - 1) * take + startIndex;
+            //var end = start + take;
+            var start = NativeContract.Ledger.CurrentIndex(_neosystem.StoreView) - (skip - 1) * take;
+            var end = start - take;
+            var lstOfBlocks = new List<Header>();
+            for (uint i = start; i > end; i--)
+            {
+                var block = NativeContract.Ledger.GetBlock(_neosystem.StoreView, i);
+                if (block == null)
+                    break;
+                lstOfBlocks.Add(block.Header);
+            }
+            if (lstOfBlocks.Count != 0)
+                return NoContent();
+            return Ok(lstOfBlocks);
+        }
+
+        /// <summary>
+        /// Gets the current block header of the connected node.
+        /// </summary>
+        /// <returns>Full Block Header Object.</returns>
+        /// <response code="200">Successful</response>
+        /// <response code="400">An error occurred. See Response for details.</response>
+        [HttpGet("blockheader/current", Name = "GetCurrnetBlockHeader")]
+        [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(Header))]
+        public IActionResult GetCurrentBlockHeader()
+        {
+            var currentIndex = NativeContract.Ledger.CurrentIndex(_neosystem.StoreView);
+            var blockheader = NativeContract.Ledger.GetHeader(_neosystem.StoreView, currentIndex);
+            return Ok(blockheader);
+        }
+
+        /// <summary>
+        /// Gets a block by an its index.
+        /// </summary>
+        /// <param name="blockIndex" example="0">Block Index</param>
+        /// <returns>Full Block Object.</returns>
+        /// <response code="200">Successful</response>
+        /// <response code="400">An error occurred. See Response for details.</response>
+        [HttpGet("blocks/{index:min(0)}", Name = "GetBlock")]
+        [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(Block))]
+        public IActionResult GetBlock(
+            [FromRoute(Name = "index")]
+            uint blockIndex)
+        {
+            var block = NativeContract.Ledger.GetBlock(_neosystem.StoreView, blockIndex);
+            if (block == null)
+                throw new BlockNotFoundException(blockIndex);
+            return Ok(block);
+        }
```

Comment:
It will be good to add confirmations here.

Reply:
confirmations?

Reply:
> confirmations?

The blocks confirmed count between this height and the latest. Although it's not in Ledger.GetBlock().
```C#
json["confirmations"] = NativeContract.Ledger.CurrentIndex(_neosystem.StoreView) - block.Index + 1;
```